cx_Freeze/initscripts/Console.py

```diff
@@ -31,7 +30,8 @@
                                                 BUILD_CONSTANTS.PYTZ_TZDATADIR)
 
 def run():
-    name, ext = os.path.splitext(os.path.basename(os.path.normcase(FILE_NAME)))
-    moduleName = "%s__main__" % name
-    code = __loader__.get_code(moduleName)
-    exec(code, {'__name__': '__main__'})
+    name = "%s__main__" % __name__[:-8]
```

Comment:
Why the [:8]? Is that to restrict to 8.3 file name restrictions from ancient file systems? Do we really need that?

Reply:
if a module name is "test', the __name__ is "test__init__", the I change it to "test__main__".

Reply:
Can we use this instead?

```
name = __name__.rpartition("__init__")[0] + "__main__"
````

I think that makes it clearer what is going on. We could probably also just use

```
name = __name__.replace("__init__", "__main__")
```

Technically someone could have a name with the string "__init__" found somewhere in there -- but that is extremely unlikely as it is bad form! Still, the rpartition() approach is reasonably understandable and doesn't have that limitation.

Reply:
the replace can be bad, if `__name__' is `"my__init__"`
Don't like slices?

Reply:
I don't mind slices. I dislike hard-coded constants. :) If you prefer, you can put a comment there as well.

Reply:
8 is the len of `"__init__"`, then -8 is the name minus it
I will update

Reply:
Yes, I understand -- after you explained it. It wasn't obvious from the code itself. Now it is. Thanks!